dpm/client/__init__.py

```diff
@@ -9,21 +9,30 @@
 class DpmException(Exception):
     pass
 
+
+class ConfigError(Exception):
+    """ The configuration passed to the client is malformed """
+    pass
+
+
 class Client(object):
 
-    def __init__(self, data_package_path='', config_path="~/.dpm/config", click=None):
+    def __init__(self, config, data_package_path='', click=None):
```

Comment:
Thinking more on the passing it datapackage in the constructor:

We will have `install` command at some point (#30). It may be useful to install multiple packages like npm\pip: `dpmpy install @pub/pak1 @pub/pak2 etc...` With current approach, bounding the client to a single dp - the cli will have to instantinate a new client for each package. This seems like one surprising API design. (Looking at any other client-like API in the wild - those usually focus on abstracting away the rest endpoint with more convenient wrappers)

The same applies to the deletion of multiple packages.

And I could imagine the command to list all your published packages on the server - it does not require a datapackage instance at all.

So I would suggest passing the datapackage to the actual method of the client, where it is needed. The multiple deletion will then look like this:
```
client = Client()
for package_id in packages:
     client.delete(package_id)
```